SmartDeviceLinkTests/DevAPISpecs/SDLPresentKeyboardOperationSpec.m

```diff
@@ -43,7 +43,14 @@
         testDelegate = OCMProtocolMock(@protocol(SDLKeyboardDelegate));
         OCMStub([testDelegate customKeyboardConfiguration]).andReturn(nil);
 
-        testInitialProperties = [[SDLKeyboardProperties alloc] initWithLanguage:SDLLanguageArSa keyboardLayout:SDLKeyboardLayoutAZERTY keypressMode:SDLKeypressModeResendCurrentEntry limitedCharacterList:nil autoCompleteList:nil];
+        testInitialProperties = [[SDLKeyboardProperties alloc] initWithLanguage:SDLLanguageArSa keyboardLayout:SDLKeyboardLayoutAZERTY keypressMode:SDLKeypressModeResendCurrentEntry limitedCharacterList:nil autoCompleteList:nil maskInputCharacters:nil customKeys:nil];
+    });
+
+    afterEach(^{
+        if (testOp) {
+            [[NSNotificationCenter defaultCenter] removeObserver:testOp];
+            testOp = nil;
+        }
```

Comment:
For what reason was this added? Were tests not passing?

Reply:
The test failed at random, usually on test server yet on my local machine it did work (zombie object received notifications and set wrong value on a next test run)

Reply:
As an option I can implement a shut down method in SDLPresentKeyboardOperation to unsubscribe from notifications or use the -finishOperation method. One more, it can subscribe for notifications more than once and receive a notification as many times as it has subscribed to it. With the current ARC memory management system a zombie object may live a few seconds after there is no references to it depending 'on the weather'. Since iOS-8 we do not need to unsubscribe from notifications explisitely and iOS does not crash any more if a zombie receives a notification and the best part of it is that the zombie still has access to data objects and can change their state.